File: csharp-reference/Characters/Broly.cs
using DragonBallArena.Models;

namespace DragonBallArena.Characters;

public sealed class Broly(string? name = null) : BaseFighter(name)
{
    public override string Id => "broly";
    public override string DisplayName => "Broly";
    public override string Greeting => "RAAAAGH!";
    public override string SignatureAttack => "Gigantic Roar";
    public override string AttackStyle => "rush";
    public override double Scale => 1.18;
    public override FighterStats Stats => new(15000, 72, 90, 0.24, 25);
    public override FighterPalette Palette => new(
        "#bd815d", "#16171c", "#6bd14b", "#5c2380", "#d3e85b", "#6cff45");
}

File: src/game/characters/Broly.ts
import { BaseFighter } from "../BaseFighter";

export class Broly extends BaseFighter {
  readonly id = "broly" as const;
  readonly name = "Broly";
  readonly greeting = "RAAAAGH!";
  readonly signatureAttack = "Gigantic Roar";
  readonly attackStyle = "rush" as const;
  readonly power = 15000;
  readonly speed = 72;
  readonly defense = 90;
  readonly criticalChance = 0.24;
  readonly baseDamage = 25;
  readonly nativeFacing = "right" as const;
  readonly scale = 1.18;
  readonly palette = {
    skin: "#bd815d",
    hair: "#16171c",
    primary: "#6bd14b",
    secondary: "#5c2380",
    accent: "#d3e85b",
    aura: "#6cff45"
  };
}
